iroha-balancer-client/src/main/kotlin/soramitsu/irohautils/balancer/client/service/IrohaBalancerClientService.kt

```diff
@@ -0,0 +1,101 @@
+package soramitsu.irohautils.balancer.client.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.rabbitmq.client.BuiltinExchangeType
+import com.rabbitmq.client.Connection
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.client.impl.DefaultExceptionHandler
+import iroha.protocol.TransactionOuterClass
+import jp.co.soramitsu.iroha.java.Utils
+import mu.KLogging
+import soramitsu.irohautils.balancer.client.config.RMQConfig
+import java.io.Closeable
+import kotlin.system.exitProcess
+
+open class IrohaBalancerClientService @JvmOverloads constructor(
+        val rmqConfig: RMQConfig,
+        private val onRmqFail: () -> Unit = {
+            logger.error { "RMQ failure. Exit." }
+            exitProcess(1)
+        }
+) : Closeable {
+
+    private val factory = ConnectionFactory()
+
+    private val objectMapper = ObjectMapper()
+
+    private val connection by lazy {
+        if (rmqConfig.username != null && rmqConfig.password != null) {
```

Comment:
if `rmqConfig.username` and `rmqConfig.password` both are null then `factory.username`  and `factory.password` won't be initialized. Maybe we make these properties strictly non-null?